### PR TITLE
Stabilize SystemChannelTypes tests

### LINKED ISSUE #5933: Nightly Builds fail due to systemwide channeltype tests
This tests only works randomly, because the installed test bundle does not provide any german translation for the battery channel...

https://ci.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/2883
```
Error Message
expected:<Niedriger Batteriestatus> but was:<null>
Stacktrace
java.lang.AssertionError: expected:<Niedriger Batteriestatus> but was:<null>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.eclipse.smarthome.core.thing.xml.test.SystemWideChannelTypesTest.systemChannelsShouldTranslateProperly(SystemWideChannelTypesTest.java:202)
```

These 2 other tests use the `ChannelTypeRegistry` which relies on the `ChannelTypeProvider`s. If the `DefaultSystemChannelTypeProvider` has not been registered, it returns empty collections for the channels.

https://ci.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/2878

```
Error Message
expected:<23> but was:<0>
Stacktrace
java.lang.AssertionError: expected:<23> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.eclipse.smarthome.core.thing.SystemWideChannelTypesTest.systemChannelTypesShouldBeAvailable(SystemWideChannelTypesTest.java:79)
```

https://ci.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/2880

```
Error Message
expected:<23> but was:<0>
Stacktrace
java.lang.AssertionError: expected:<23> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.eclipse.smarthome.core.thing.SystemWideChannelTypesTest.systemChannelTypesShouldBeTranslatedProperly(SystemWideChannelTypesTest.java:109)
```

Testing the `DefaultSystemChannelTypeProvider` directly obviously makes more sense.

### PR DESCRIPTION
**Pull request by triller-telekom**
Fixes #5933

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/SystemChannels.bundle/ESH-INF/i18n/SystemChannels_de.properties [hunk 1]
```diff
@@ -1,5 +1,7 @@
 channel-type.system.my-channel.label = Mein String My Channel
 channel-type.system.my-channel.description = Wetterinformation mit My Channel Type Beschreibung
 
+channel-type.system.low-battery.label = Niedriger Batteriestatus
```
Comment (cweitkamp) [inline bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/SystemChannels.bundle/ESH-INF/i18n/SystemChannels_de.properties]: I am not sure if this is what we want to do here. When defining a things channel using a default system channel type we expect that the label and description are inherited and so the translations for them.

Does it maybe makes more sense to switch from `ChannelTypeRegistry` to `ChannelTypeProvider` here as well? wdyt?

Comment (triller-telekom) [reply bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/SystemChannels.bundle/ESH-INF/i18n/SystemChannels_de.properties]: Good point, thanks. I have changed it.

**Comment by triller-telekom** [comment]
@maggu2810 Could you please have a look at this PR that aims on fixing our [nightly build](https://ci.eclipse.org/smarthome/)?

**Comment by triller-telekom** [comment]
FYI: Follow up is #5945

**Pull request merged**